go-controller/pkg/node/node.go

```diff
@@ -600,15 +605,103 @@ func (n *OvnNode) WatchEndpoints() error {
 	return err
 }
 
+// WatchNodes will watch all nodes. When the current node is updated, it will write a new encap IP
+// to OVS in case that the primary interface IP changed and in case  the encap IP is not overwritten by
+// configuration.
+func (n *OvnNode) WatchNodes() error {
+	_, err := n.watchFactory.AddNodeHandler(cache.ResourceEventHandlerFuncs{
```

Comment:
do we really need a watcher/handler to do this? We already have the node ip address manager that will update the annotation when the addresses change, why cant it also just update the encap ip?